engines/osxnative/sandbox.go

```diff
@@ -136,21 +125,99 @@ func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
 	cmd := exec.Command(s.taskPayload.Command[0], s.taskPayload.Command[1:]...)
 	cmd.Stdout = stdoutLogWriter{s.context}
 	cmd.Stderr = stderrLogWriter{s.context}
+
+	// USER and HOME are treated seperately because their values
+	// depend on either we create the new user successfuly or not
+	processUser := os.Getenv("USER")
+	processHome := os.Getenv("HOME")
+
+	// If we fail to create a new user, the most probable cause is that
+	// we don't have enough permissions. Chances are that we are running
+	// in in a development environment, so do not fail the task to tests
+	// run successfully.
+	u := user{}
+	if err = u.create(); err != nil {
```

Comment:
As we talked earlier, I implemented a environment variable based configuration. If `TASKCLUSTER_WORKER_ENV=Production` (non case sensitive), the task fails if it can't create the user.